### PR TITLE
Enable optional TensorList fallback

### PR DESCRIPTION
**Pull request by qqaatw**
Previously only `at::Tensor` and `at::TensorList` are able to fallback to cpu, while other arguments are kept untouched.

This PR enables `c10::List<c10::Optional<at::Tensor>>` fallback so that the tensors on a device in optional list can fallback to cpu.

Operators supported: `index_put` and `index_out`.

Partially resolve. #81051

### CODE REVIEW

File: aten/src/ATen/native/CPUFallback.cpp [hunk 1]
```diff
@@ -77,6 +77,20 @@ void cpu_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
     if (ivalue.isTensor()) {
       tensor_args.push_back(ivalue.toTensor());
       tensor_args_indices.push_back(idx);
+    } else if (ivalue.isOptionalTensorList()) {
+      auto ivalue_vec = ivalue.toOptionalTensorList().vec();
+      c10::List<at::Tensor> result;
+
+      result.reserve(ivalue_vec.size());
+      for (const auto i : c10::irange(ivalue_vec.size())) {
+        if (ivalue_vec[i].has_value()){
```
Comment (qqaatw) [inline aten/src/ATen/native/CPUFallback.cpp:86]: Not sure if we have to handle no value.

**Comment by qqaatw** [comment]
@bdhirsh Can you please open and review this one? thank you.

**Comment by bdhirsh** [comment]
@qqaatw sorry for missing this one. I can't re-open the PR, since `master` was recently changed to `main`. If you open a fresh PR against the main branch, I'll happily stamp!

**Comment by qqaatw** [comment]
@bdhirsh If the new PR will surely be reviewed, I'm happy to submit one. (sorry, I know you guys are busy but I'm tired with pinging reviewers again and again).

**Pull request closed without merging**